app/routes/user/password.tsx

```diff
@@ -0,0 +1,140 @@
+  const [confirmPassword, setConfirmPassword] = useState('')
+  const [isNewPasswordTouched, setIsNewPasswordTouched] = useState(false)
+  const [isConfirmPasswordTouched, setIsConfirmPasswordTouched] =
+    useState(false)
+  const passwordsMatch = newPassword === confirmPassword
+  const passwordsAreEmpty = newPassword === '' && confirmPassword === ''
+  const shouldDisableSubmit =
+    passwordsAreEmpty || !isNewPasswordTouched || !passwordsMatch
+  const isMatchError =
+    !isConfirmPasswordTouched || passwordsMatch ? '' : 'usa-input--error'
+
+  const valid =
+    !startsOrEndsWithWhitespace(newPassword) && checkPassword(newPassword)
+  const isValidPassword =
+    !isNewPasswordTouched || valid ? '' : 'usa-input--error'
+
+  return (
+    <>
+      <h1>Reset Password</h1>
+      <>
+        <fetcher.Form method="POST">
+          <Label htmlFor="oldPassword">Old Password</Label>
+          <TextInput
+            data-focus
+            name="oldPassword"
+            id="oldPassword"
+            type="password"
+            placeholder="Old Password"
+          />
+          <Label htmlFor="newPassword">New Password</Label>
+          <TextInput
+            className={isValidPassword}
+            name="newPassword"
+            id="newPassword"
+            type="password"
+            placeholder="New Password"
+            onChange={(e) => {
+              setNewPassword(e.target.value)
+              setIsNewPasswordTouched(true)
+            }}
+            value={newPassword}
+          />
+          <Label htmlFor="confirmPassword">Retype New Password</Label>
+          <TextInput
+            className={isMatchError}
+            name="confirmPassword"
+            id="confirmPassword"
+            type="password"
+            placeholder="Retype New Password"
+            onChange={(e) => {
+              setConfirmPassword(e.target.value)
+              setIsNewPasswordTouched(true)
+              setIsConfirmPasswordTouched(true)
+            }}
+            value={confirmPassword}
+          />
+
+          <Button
+            className="margin-y-2"
+            disabled={shouldDisableSubmit}
+            type="submit"
+          >
+            Reset Password
+          </Button>
+          {fetcher.state !== 'idle' && (
+            <>
+              <Spinner /> Saving...
+            </>
+          )}
+          {fetcher.state === 'idle' && fetcher.data !== undefined && (
+            <>
+              <Icon.Check color="green" /> Saved
+            </>
+          )}
+        </fetcher.Form>
+      </>
+      <h3 className="site-preview-heading margin-0">
+        New password must contain:
+      </h3>
+      <ul className="usa-list">
+        <li>A lower case letter</li>
+        <li>An upper case letter</li>
+        <li>A number</li>
+        <li>At least 8 characters</li>
+        <li>At least 1 special character or space</li>
+        <li>No leading or trailing spaces</li>
```

Comment:
How do we know that this matches the password requirements of our Cognito user pool?

Reply:
<img width="706" alt="Screen Shot 2023-07-03 at 2 42 32 PM" src="https://github.com/nasa-gcn/gcn.nasa.gov/assets/6545874/ae15c8dc-1394-45f5-95bb-d7eaf68a1cd3">
I matched what is specified in the login.

Reply:
Ah, clever.